Generate the mocking framework (a bit)

DrDeano opened the pull request:
Instead of adding new types in 5 different places, this adds a generation tool that you add the type once and it will be populated in the framework for you.
This is defo one of those over engineered projects, but I had fun making it.

SamTebbs33 commented on test/gen_types.zig:
```diff
@@ -0,0 +1,260 @@
+fn genGenericFunc(comptime intermediate: []const u8, comptime trail: []const u8, comptime end: []const u8) []const u8 {
+    comptime var str: []const u8 = "";
+    inline for (types) |t, i| {
+        if (i == 0) {
+            // Want nice formatting
+            str = str ++ "    " ++ t[0] ++ intermediate ++ t[1] ++ trail;
+        } else {
+            str = str ++ "                " ++ t[0] ++ intermediate ++ t[1] ++ trail;
+        }
```
We could probably remove the duplicate code by assigning the big ol' block of spaces to a const and using that.

SamTebbs33 reviewed (changes requested):
This is indeed one of the most beautifully over-engineered solutions I have ever seen :D

SamTebbs33 commented on test/gen_types.zig:
```diff
@@ -145,12 +147,12 @@ fn genGenericFunc(comptime intermediate: []const u8, comptime trail: []const u8,
-            str = str ++ "                " ++ t[0] ++ intermediate ++ t[1] ++ trail;
+            str = str ++ tab_space ** 4 ++ t[0] ++ intermediate ++ t[1] ++ trail;
         }
     }
```
I meant something a bit more like
```
const spaces = if (i == 0) " " ** 4 else " " ** 16;
str = str ++ spaces ++ t[0] ++ intermediate ++ t[1] ++ trail;
```

DrDeano replied:
Sure

SamTebbs33 reviewed (approved):
lgtm

DrDeano committed:
Generate the mocking framework (a bit)

Instead of adding new types in 5 different places, this adds a generation tool that you add the type once and it will be populated in the framework for you.
This is defo one of those over engineered projects, but I had fun making it.

Replaces empty spaces with const


Refactor spaces
